addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -715,6 +731,10 @@ import wUtils from '@website/js/utils';
                     return !(value >= comparable && value <= between);
                 case 'equal or after':
                     return value >= comparable;
+                case "lessyears":
+                    currentDate.setFullYear(currentDate.getFullYear() - comparable);
+                    value = new Date(value * 1000);
+                    return value > currentDate;
```

Comment:
Previously, `currentDate` contained today's date, but now it contains the date from a `comparable` number of years ago.
You can introduce new variable here to reduce the confusion. or simple 
create a variable something like,

`dateDiff = currentDate - value;
return dateDiff < comparable;`